app/models/reward.rb

```diff
@@ -5,4 +5,6 @@ class Reward < ApplicationRecord
   validates :title, :description, :price, presence: true
 
   has_many :orders, dependent: :nullify
+  has_many :category_rewards, dependent: :nullify
```

Comment:
` dependent: nullify `it will set, `reward_id` in category_reward to `nil`, but in your db you have a not null restrictions